contracts/vaults/TreasuryVault/RibbonTreasuryVault.sol

```diff
@@ -273,7 +273,8 @@ contract RibbonTreasuryVault is
         );
         require(_initParams._premiumAsset != address(0), "!_premiumAsset");
         require(
-            _initParams._whitelist.length <= WHITELIST_LIMIT,
+            _initParams._whitelist.length <= WHITELIST_LIMIT &&
+                _initParams._whitelist.length > 0,
             "!_whitelist"
```

Comment:
These can be split out into 2 different revert messages so that it's clearer

`!_whitelist` and `Whitelist exceed limit`